packages/template-retail-react-app/app/pages/product-detail/index.jsx

```diff
@@ -34,27 +34,52 @@ import RecommendedProducts from '../../components/recommended-products'
 import ProductView from '../../partials/product-view'
 
 // Others/Utils
-import {HTTPNotFound} from 'pwa-kit-react-sdk/ssr/universal/errors'
+// import {HTTPNotFound} from 'pwa-kit-react-sdk/ssr/universal/errors'
 
 // constant
 import {
     API_ERROR_MESSAGE,
-    MAX_CACHE_AGE,
+    // MAX_CACHE_AGE,
     TOAST_ACTION_VIEW_WISHLIST,
     TOAST_MESSAGE_ADDED_TO_WISHLIST
 } from '../../constants'
 import {rebuildPathWithParams} from '../../utils/url'
-import {useHistory} from 'react-router-dom'
+import {useHistory, useParams, useLocation} from 'react-router-dom'
 import {useToast} from '../../hooks/use-toast'
+import {useCategory, useProduct, useShopperBasketsMutation} from 'commerce-sdk-react'
+import {useBasket as useBasketHook} from '../../hooks/useBasket'
 
-const ProductDetail = ({category, product, isLoading}) => {
+const ProductDetail = () => {
     const {formatMessage} = useIntl()
-    const basket = useBasket()
+    // const basket = useBasket()
     const history = useHistory()
+    const location = useLocation()
     const einstein = useEinstein()
-    const variant = useVariant(product)
     const toast = useToast()
     const navigate = useNavigation()
+
+    //===================== Commerce react hooks==========================//
+    // **************Basket******************** //
+    const {baskets, hasBasket} = useBasketHook()
+    const createBasket = useShopperBasketsMutation('createBasket')
+    const addItemToBasket = useShopperBasketsMutation('addItemToBasket')
+
+    // ************************** Product and category ************************//
+    //TODO: Fix loading problem when fetching variants
+    const {productId} = useParams()
+    const urlParams = new URLSearchParams(location.search)
+    const {data: product, isLoading: isProductLoading, error: productError} = useProduct({
+        id: urlParams.get('pid') || productId,
+        allImages: true
+    })
+    // only fetch category on master product
+    const {data: category, error: categoryError} = useCategory(
+        {id: product?.primaryCategoryId, levels: 1},
+        {enabled: !!product?.primaryCategoryId}
```

Comment:
FYI dependent query like this is currently not supported in SSR, but it works on client-side. See this [Slack](https://sfcc-unofficial.slack.com/archives/C02KUCHDEKH/p1667306311787719?thread_ts=1667248680.487539&cid=C02KUCHDEKH) conversation.